app/models/async_transaction/vet360/base.rb

```diff
@@ -72,12 +79,41 @@ def self.update_transaction_from_api(transaction_record, api_response)
         transaction_record
       end
 
-      # Returns true if a transaction is "over"
-      # @return [Boolean]
+      # Returns true or false if a transaction is "over"
+      # @return [Boolean] true if status is "over"
       def finished?
         # These SHOULD go hand-in-hand...
         FINAL_STATUSES.include?(transaction_status) || status == COMPLETED
       end
+
+      # Wrapper for .refresh_transaction_status which finds any outstanding transactions
+      #   for a user and refreshes them
+      # @param user [User] the user whose transactions we're checking
+      # @params service [Vet360::ContactInformation::Service] an initialized vet360 client
+      # @return [Array] An array with any outstanding transactions refreshed. Empty if none.
+      def self.refresh_transaction_statuses(user, service)
+        ongoing_transactions = last_ongoing_transactions_for_user(user)
+        refreshed_transactions = []
+        ongoing_transactions.each do |transaction|
+          refreshed_transactions << refresh_transaction_status(
+            user,
+            service,
+            transaction.transaction_id
+          )
+        end
+        return refreshed_transactions
```

Comment:
There is this pretty cool method called [`each_with_object`](https://ruby-doc.org/core-2.3.1/Enumerator.html#method-i-each_with_object) that allows you to forgo the assigning of an outside array variable, and then the explicit returning of that array outside of the loop.

For example:

```ruby
[1, 2, 3, 4].each_with_object([]) do |num, the_returned_array|
  the_returned_array << num * 2
end
=> [2, 4, 6, 8]
```

So if you wanted to, you could eliminate lines 96 and 104 (and 95 too, with that other comment).  Just an option.